### PR TITLE
Update pre-base image

### PR DESCRIPTION
**Pull request by patrick-5546**
### Description

Some upstream changes have broken pre-base.Dockerfile, specifically being able to build OMPL. In this PR, I'm updating OMPL to the latest commit and manually adding the changes made to [ompl.Dockerfile](https://github.com/ompl/ompl/blob/main/scripts/docker/ompl.Dockerfile) since I last looked at it.

### Verification

- [x] Rebuild all images with these changes

### Resources

- Commit history of OMPL: https://github.com/ompl/ompl/commits/main/
- Commit history of ompl.Dockerfile: https://github.com/ompl/ompl/commits/main/scripts/docker/ompl.Dockerfile
- Building pre-base: https://github.com/UBCSailbot/sailbot_workspace/blob/main/.devcontainer/pre-base/README.md#how-to-build
- Building base, local-base, and dev: https://github.com/UBCSailbot/sailbot_workspace/blob/main/.devcontainer/base-dev/README.md#how-to-build

### CODE REVIEW

**Comment by patrick-5546** [comment]
@SPDonaghy @stevenxu27 can you try to build the pre-base and base-dev images following the instruction in the READMEs? @SPDonaghy you don't have to build the base-dev one because you did it before. Reference: https://ubcsailbot.github.io/docs/latest/current/sailbot_workspace/docker_images/

If you have any questions comment them on this PR. Once you're successful please approve, thanks

**Comment by SPDonaghy** [comment]
@patrick-5546 I am just looking at the README for the pre-base image and I don't think my computer has the resources to build the image. 

Just so I understand better, why does it need to be run on a personal computer?

**Comment by patrick-5546** [comment]
> I am just looking at the README for the pre-base image and I don't think my computer has the resources to build the image.

Maybe overnight can you try building it to see if it works? I'm just curious, I dynamically limited the number of processes building OMPL uses so hopefully it doesn't crash. But just in case save and close other all other programs before building lol.

> Just so I understand better, why does it need to be run on a personal computer?

Before it was because the VM that GitHub Actions ran on was even worse than most personal computers. However that seems to have changed, it seems like the runners now have 4 processors and 16 GB of RAM. Let me try to run this in CI, I'll keep you updated.

One more thing I should have mentioned, the script should be run in the Ubuntu terminal (not in the dev container)

**Comment by SPDonaghy** [comment]
Got it, okay I will try to run it overnight and give an update tomorrow.

**Comment by patrick-5546** [comment]
> Let me try to run this in CI

It took 2 hours and 45 minutes to build from scratch, which isn't too bad I guess: https://github.com/UBCSailbot/sailbot_workspace/actions/runs/7689182429

However, as maintainers-to-be of sailbot workspace, I still think you should at least try going through the [updated] debugging steps: https://github.com/UBCSailbot/sailbot_workspace/tree/update-pre-base/.devcontainer/base-dev#debugging-locally

File: .devcontainer/base-dev/base-dev.Dockerfile
Comment (SPDonaghy) [inline .devcontainer/base-dev/base-dev.Dockerfile:1]: I see that you combined pre-base and base-dev images. So in the future, when members add a dependency to the base-dev image, the workflow to build the image will install ompl and ros each time right?

**Review by SPDonaghy — COMMENTED** [review]
Just testing out the review process.

File: .devcontainer/base-dev/base-dev.Dockerfile
Comment (patrick-5546) [reply .devcontainer/base-dev/base-dev.Dockerfile:1]: Technically yes, but in reality docker caches each command in the dockerfile (creates a layer for each command). So ideally Docker will start executing commands at the first line that you change.

**Comment by stevenxu27** [comment]
I ran into some type of error last night (I think something went wrong with my laptop haha) but I'm rebuilding right now.

**Comment by patrick-5546** [comment]
> I think something went wrong with my laptop

Hopefully it wasn't the build crashing your laptop by using too much memory

**Comment by stevenxu27** [comment]
Hey Patrick! I ran the Create New Driver and Build Command from [https://github.com/UBCSailbot/sailbot_workspace/tree/update-pre-base/.devcontainer/base-dev#debugging-locally](url) but I was a bit confused about how to proceed further. Was that all I needed to move on to building the base-dev images?

**Comment by patrick-5546** [comment]
> Was that all I needed to move on to building the base-dev images?

If the build command completed successfully then you have already built all the images and you're good to go. No need to continue on and push the images, and you can approve this PR.

In the work party this week I plan to explain to y'all what each of the commands are doing, please bring any other questions you have as well. Then we can discuss what you can work on next

**Comment by stevenxu27** [comment]
Sounds good I think it did build correctly. I was checking with Sean about that and it seems we have the same thing. Looking forward to learning more this weekend!

**Review by stevenxu27 — APPROVED** [review]

**Comment by SPDonaghy** [comment]
I also had it build successfully last night. Took 2 hrs 7 mins

**Pull request merged**
